bitnami/odoo/templates/svc.yaml

```diff
@@ -16,20 +16,23 @@ metadata:
     {{- end }}
 spec:
   type: {{ .Values.service.type }}
-  sessionAffinity: {{ .Values.service.sessionAffinity }}
   {{- if and .Values.service.clusterIP (eq .Values.service.type "ClusterIP") }}
   clusterIP: {{ .Values.service.clusterIP }}
   {{- end }}
-  {{- if (or (eq .Values.service.type "LoadBalancer") (eq .Values.service.type "NodePort")) }}
+  {{- if or (eq .Values.service.type "LoadBalancer") (eq .Values.service.type "NodePort") }}
   externalTrafficPolicy: {{ .Values.service.externalTrafficPolicy | quote }}
   {{- end }}
-  {{- if eq .Values.service.type "LoadBalancer" }}
-  {{- if .Values.service.loadBalancerSourceRanges }}
-  loadBalancerSourceRanges: {{- toYaml .Values.service.loadBalancerSourceRanges | nindent 4 }}
+  {{- if and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerSourceRanges)) }}
+  loadBalancerSourceRanges: {{ .Values.service.loadBalancerSourceRanges }}
   {{- end }}
-  {{- if not (empty .Values.service.loadBalancerIP) }}
+  {{- if and (eq .Values.service.type "LoadBalancer") (not (empty .Values.service.loadBalancerIP)) }}
```

Comment:
We could use just one `if` for `eq .Values.service.type "LoadBalancer"` wrapping all this content instead of checking it on every parameters

Reply:
The service section is copied from the template: https://github.com/bitnami/charts/blob/ff8392dfb4f0771b17128c8792c883e06ea7c14b/template/CHART_NAME/templates/service.yaml#L21-L39